### PR TITLE
[Test][E2E] Add thread leak check for connector

### PR DESCRIPTION
**Pull request by Hisoka-X**
### Purpose of this pull request
This PR add thread leak check for connector. We should make sure our connector stable on cluster mode, does not leak thread resource.

**This PR does not solve all problems, it just reduces the probability of memory overflow.**

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
add new test.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
good idea

**Comment by Hisoka-X** [comment]
cc @EricJoy2048 @liugddx @hailin0

**Comment by Hisoka-X** [comment]
This PR does not solve all problems, it just reduces the probability of memory overflow.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 1]
```diff
@@ -276,6 +279,24 @@ private void handleJobInfoById(HttpGetCommand command, String uri) {
         }
     }
 
+    private void getRunningThread(HttpGetCommand command) {
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:282]: Does this rest api need to be maintained in documentation?

**Review by liugddx — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**